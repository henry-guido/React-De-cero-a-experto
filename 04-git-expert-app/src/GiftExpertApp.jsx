import { useState } from 'react';
import {AddCategory, GifGrid} from './components'

export const GiftExpertApp = () => {
  const [categories, setCategories] = useState(['one punch']);

  const onAddCategory = (val) => {
    setCategories([val, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onAddCategory={onAddCategory} />
      {categories.map((category) => (
        <div key={category} className='card-grid'>
          <GifGrid  category={category} />
        </div>
      ))}
    </>
  );
};
