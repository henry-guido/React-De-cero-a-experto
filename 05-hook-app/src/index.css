body{
  padding: 20px;
  background-color: #444;
  filter: invert();
}

button{
  margin-right: 10px;
}

nav{
  display: flex;
  justify-content: space-between;
}

ul{
  display: flex;
  align-items: center;
  margin: 0;
  gap: 10px
}

li{
  list-style: none;
}

a{
  text-decoration: none;
}