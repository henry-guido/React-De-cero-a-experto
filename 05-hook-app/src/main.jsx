import ReactDOM from 'react-dom/client';
import { BrowserRouter} from 'react-router-dom'
// import HookApp from './HookApp'
// import { CounterApp } from './01-useState/CounterApp'

import './index.css';
import { Padre } from './07-tarea-memo/Padre';
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks';
// import { FocusScreen } from './04-useRef/FocusScreen';
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
// import {Layout} from './05-useLayoutEffect/Layout'
// import './08-useReducer/intro-reducer'
// import { TodoApp } from './08-useReducer/TodoApp';
import { MainApp } from './09-useContext/MainApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
   {/* <React.StrictMode> */}
  <MainApp />
  {/* </React.StrictMode>, */}
  </BrowserRouter>
);
