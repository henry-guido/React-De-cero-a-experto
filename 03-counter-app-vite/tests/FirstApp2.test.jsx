import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Prueba en FirstApp', () => {
  const title = 'Hola, soy un titulo';
  const subTitle = 'Hola, soy un subtitulo';

  test('debe hacer match con snapshot', () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('debe mostrar Hola, soy un titulo ', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('Debe buscar titulo en H1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
  });

  test('Debe de mostrar el subtitulo enviado por props', () => { 
    render(<FirstApp title={title} subTitle={subTitle}/>);
    expect(screen.getAllByText(subTitle).length).toBe(1)
   })
});
