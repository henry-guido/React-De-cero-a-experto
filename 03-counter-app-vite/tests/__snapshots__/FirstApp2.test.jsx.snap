// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en FirstApp debe hacer match con snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy un titulo
     
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
